enterprise_access/apps/content_assignments/api.py

```diff
@@ -392,14 +393,22 @@ def _try_populate_assignments_lms_user_id(assignments):
     for assignment_chunk in chunks(assignments_with_empty_lms_user_id, USER_EMAIL_READ_BATCH_SIZE):
         emails = [assignment.learner_email for assignment in assignment_chunk]
         # Construct a list of tuples containing (email, lms_user_id) for every assignment in this chunk.
-        email_lms_user_id = User.objects.filter(
-            email__in=emails,  # this is the part that could exceed max statement length if batch size is too large.
-            lms_user_id__isnull=False,
-        ).values_list('email', 'lms_user_id')
+        # this is the part that could exceed max statement length if batch size is too large.
+        # There's no case-insensitive IN query in Django, so we have to build up a big
+        # OR type of query.
+        email_filter = Q()
+        for assignment_email in emails:
+            email_filter |= Q(email__iexact=assignment_email)
```

Comment:
FWIW, I think this would increase the statement length (adding a string of OR email checks instead of a flat list of emails).  Might be worth halving USER_EMAIL_READ_BATCH_SIZE to 175 just to be safe.

Reply:
Tuned it down to a nice even 100.